firestore.rules

```diff
@@ -18,7 +18,7 @@ service cloud.firestore {
       allow write: if request.auth.uid != null;
     }
     match /users/{userId} {
-      allow read: if true;
+      allow read: if request.auth.uid != null;
```

Comment:
@mike-lvov Have you tested what impacts this has to functionality on the site (if any). I noticed weird things break in ways I wouldn’t have expected when I tried changing some of the other rules recently.

Also, there are other rules here that should be similarly fixed up at the same time, so should really be fixing those as well. Let’s clean all of this up properly in one fell swoop.

Reply:
@0xdevalias I didn't do the testing, but will do it before merging it in

Reply:
Did the testing

Reply:
See https://github.com/sparkletown/sparkle/pull/1854#pullrequestreview-712273168